AK Apolo/app/src/main/java/ayds/apolo/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,157 @@
+package ayds.apolo.songinfo.moredetails.fulllogic
+
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.View
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.text.HtmlCompat
+import androidx.core.text.HtmlCompat.FROM_HTML_MODE_LEGACY
+import ayds.apolo.songinfo.R
+import ayds.apolo.songinfo.moredetails.fulllogic.DataBase.Companion.getInfoArtist
+import ayds.apolo.songinfo.moredetails.fulllogic.DataBase.Companion.getUrl
+import ayds.apolo.songinfo.moredetails.fulllogic.DataBase.Companion.saveArtist
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import com.squareup.picasso.Picasso
+import retrofit2.Retrofit
+import retrofit2.converter.scalars.ScalarsConverterFactory
+import java.util.*
+
+private const val URL_SERVICE = "https://ws.audioscrobbler.com/2.0/"
+private const val LASTFM_LOGO_IMAGE_URL =
+    "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d4/Lastfm_logo.svg/320px-Lastfm_logo.svg.png"
+private const val LOCALLY_STORED_MARKER = "[*]"
+private const val ARTIST_KEY = "artist"
+private const val URL_KEY = "url"
+private const val BIO_KEY = "bio"
+private const val CONTENT_KEY = "content"
+private const val HTML_DOC_WIDTH = 400
+private const val HTML_DOC_FONT = "arial"
+private const val NO_RESULTS = "No results"
+
+class OtherInfoWindow : AppCompatActivity() {
+    private lateinit var viewArtisInfo: TextView
+    private lateinit var artistInfoContent: String
+    private lateinit var artistFullArticleURL: String
+    private val dataBase = DataBase(this)
+    private val lastFMAPI = createConnectionWithService()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_other_info)
+        viewArtisInfo = findViewById(R.id.textPane2)
+        getArtistData()
+        setUpFullArticleButtonURL()
+    }
+
+    private fun getArtistData() {
+        Thread {
+            intent.getStringExtra(ARTIST_NAME_EXTRA)?.let {
```

Comment:
Usar un método aparte para obtener el nombre del artista.

```suggestion
        val artistName = getArtistName()
        Thread {
            val artistInfo = getArtistInfo(name)
             showMoreDetails(artistInfo)    
        }
```